AdobeStockAsset/view/adminhtml/templates/system/config/testconnection.phtml

```diff
@@ -6,13 +6,23 @@
 
 /** @var $block \Magento\AdobeStockAsset\Model\Block\Adminhtml\System\Config\TestConnection */
 ?>
-<button class="scalable" type="button" id="<?= $block->escapeUrl($block->getData('html_id')) ?>" data-mage-init='{"testConnection":{
-    "url": "<?= $block->escapeHtmlAttr($block->getData('ajax_url')) ?>",
-    "elementId": "<?= $block->escapeHtmlAttr($block->getData('html_id')) ?>",
-    "successText": "<?= $block->escapeHtmlAttr(__('Successful! Test again?')) ?>",
-    "fieldToCheck": "success",
-    "failedText": "<?= $block->escapeHtmlAttr(__('Connection failed! Test again?')) ?>", "validation": {}}}'>
-    <span id="<?= $block->escapeHtmlAttr($block->getData('html_id')) ?>_result">
+<div id="<?= $block->escapeHtmlAttr($block->getApiKeyValidationResultContainerId()) ?>" class=""></div>
+<button class="scalable" type="button" id="<?= $block->escapeHtmlAttr($block->getHtmlId()) ?>">
+    <span id="<?= $block->escapeHtmlAttr($block->getHtmlId()) ?>_result">
         <?= $block->escapeHtml($block->getData('button_label')) ?>
     </span>
 </button>
+<script type="text/x-magento-init">
```

Comment:
I'd try to apply a Magento UI components approach for rendering this button. To do this it's needed to:

1. Move button HTML with knockout annotations to a .html template file, i.e.

         <div data-bind="text: message"></div>
         <button data-bind="click: testConnection"><span data-bind="text: buttonLabel"></span></button>

2. Create a component js responsible for providing data for a template, i.e.:

        define([
        'ko',
        'uiComponent',
        'jquery'
        ], function (ko, Component, $) {
        'use strict';

        return Component.extend({
            defaults: {
                buttonLabel: '',
                message: '',
            },
        /**
         * @public
         */
        testConnection: function () {
            $.ajax({type: "POST",
    ....

3. Initialize the component from the phtml template

        <div data-bind="scope: 'testAdobeStockConnection'">
        <!-- ko template: getTemplate() --><!-- /ko -->
        </div>
        <script type="text/x-magento-init">
        {
            "*": {
                "Magento_Ui/js/core/app": {
                    "components": {
                        "testAdobeStockConnection": {
                            "component": "Magento_AdobeStockAsset/js/connection",
                            "template" : "Magento_AdobeStockAsset/connection"
                        }
                    }
                }
            }
        }
        </script>


Disclaimer: The code provides in these examples have never been tested :)